test/controllers/api_users/permissions_controller_test.rb

```diff
@@ -123,21 +123,25 @@ class ApiUsers::PermissionsControllerTest < ActionController::TestCase
       assert_not_authenticated
     end
 
-    should "redirect once the permissions have been updated" do
-      application = create(:application, with_non_delegatable_supported_permissions: %w[new old])
-      api_user = create(:api_user, with_permissions: { application => %w[old] })
+    should "update non-signin permissions, retaining the signin permission, then redirect to the API applications path" do
+      application = create(:application)
+      old_permission = create(:supported_permission, application:)
+      new_permission = create(:supported_permission, application:)
+
+      api_user = create(:api_user,
+                        with_signin_permissions_for: [application],
```

Comment:
Interesting that the grantee didn't previously have the signin permission in this test. Perhaps you can technically update users' non-signin permissions without them having access to the app